test/test_script/test_external_m3u_playlist.cc

```diff
@@ -0,0 +1,176 @@
+  EXPECT_CALL(*commonScriptMock, getLastPath(Eq("/location/of/playlist.m3u"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, createContainerChain(ElementsAre("Playlists", "Directories", "of", "Playlist Title"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, print(Eq("Unknown playlist mimetype: 'no/type' of playlist '/location/of/playlist.m3u'"))).WillOnce(Return(1));
+
+  addGlobalFunctions(ctx, js_global_functions);
+  dukMockPlaylist(ctx, "Playlist Title", "/location/of/playlist.m3u", "no/type");
+
+  executeScript(ctx);
+}
+
+TEST_F(ExternalUrlM3UPlaylistTest, AddsCdsObjectFromM3UPlaylistWithExternalUrlPlaylistAndDirChains) {
+  const string MIME_TYPE_AUDIO_MPEG = "audio/mpeg";
+  const string OBJTYPE_ITEM_EXTERNAL_URL = "8";
+
+  map<string, string> asPlaylistChain = {
+      make_pair("title", "(#1 - 177/750) : Ibiza Global Radio :"),
+      make_pair("location", "http://46.105.171.217:8024"),
+      make_pair("mimetype", MIME_TYPE_AUDIO_MPEG),
+      make_pair("objectType", OBJTYPE_ITEM_EXTERNAL_URL),
+      make_pair("playlistOrder", "1"),
+      make_pair("protocol", "http-get"),
+      make_pair("description", "Song from Playlist Title"),
+      make_pair("upnpclass", UPNP_DEFAULT_CLASS_MUSIC_TRACK)
+  };
+
+  map<string, string> asPlaylistDirChain = {
+      make_pair("title", "(#1 - 177/750) : Ibiza Global Radio :"),
+      make_pair("location", "http://46.105.171.217:8024"),
+      make_pair("mimetype", MIME_TYPE_AUDIO_MPEG),
+      make_pair("objectType", OBJTYPE_ITEM_EXTERNAL_URL),
+      make_pair("playlistOrder", "2"),
+      make_pair("protocol", "http-get"),
+      make_pair("description", "Song from Playlist Title"),
+      make_pair("upnpclass", UPNP_DEFAULT_CLASS_MUSIC_TRACK)
+  };
+
+  // Expecting the common script calls..and will proxy through the mock objects for verification.
+  EXPECT_CALL(*commonScriptMock, getPlaylistType(Eq("audio/x-mpegurl"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, print(Eq("Processing playlist: /location/of/playlist.m3u"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, createContainerChain(ElementsAre("Playlists", "All Playlists", "Playlist Title"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, getLastPath(Eq("/location/of/playlist.m3u"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, createContainerChain(ElementsAre("Playlists", "Directories", "of", "Playlist Title"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, readln(Eq("#EXTM3U"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, readln(Eq("#EXTINF:-1,(#1 - 177/750) : Ibiza Global Radio :"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, readln(Eq("http://46.105.171.217:8024"))).WillOnce(Return(1));
+  EXPECT_CALL(*commonScriptMock, readln(Eq("-EOF-"))).WillOnce(Return(0));
+  EXPECT_CALL(*commonScriptMock, addCdsObject(IsIdenticalMap(asPlaylistChain),
```

Comment:
The entire purpose of any of these scripts is to call the `addCdsObject` c++ function.  So we monitor it quite closely.